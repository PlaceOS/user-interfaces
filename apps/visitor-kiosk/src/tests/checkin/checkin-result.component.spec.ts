import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';
import { of } from 'rxjs';

import { CheckinResultsComponent } from '../../app/checkin/checkin-result.component';
import { CheckinStateService } from '../../app/checkin/checkin-state.service';

describe('CheckinComponent', () => {
    let spectator: SpectatorRouting<CheckinResultsComponent>;
    const createComponent = createRoutingFactory({
        component: CheckinResultsComponent,
        providers: [
            {
                provide: CheckinStateService,
                useValue: { event: of({}), guest: of({}) },
            },
        ],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });
});
