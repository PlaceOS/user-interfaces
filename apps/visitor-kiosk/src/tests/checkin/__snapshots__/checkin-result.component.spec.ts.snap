// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckinComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="bg-white rounded shadow overflow-hidden relative flex flex-col items-center my-4 p-4 mx-auto space-y-4"
  >
    <h3
      class="text-xl"
    >
      You are checked in!
    </h3>
    <p
      class="text-center"
    >
       Welcome, you have a meeting at  with .
      <br />
        has been notified and will be with you shortly.Â  
    </p>
    <p>
       Please wait in the lobby. 
    </p>
    
    
    <a
      button=""
      class="w-32"
      mat-button=""
    >
       Done 
    </a>
  </div>
  
</div>
`;
